models/base.py

```diff
@@ -1933,17 +1940,17 @@ def update_downstream_badness(self, session=None, commit=True):
         """
         # make sure this object is current:
         with SmartSession(session) as session:
+            merged_self = session.merge(self)
```

Comment:
I haven't noticed this syntax before, but looking at the documentation it seems that this gives you the proper instance that is in the session, as opposed to `self` which remains outside of the session after this line. Is that correct?

Reply:
Yes you get a brand new object, without changing the original. Note that the new object loses all the attributes it had that are not database columns (so if you have something like `_data` it will be back to `None` on the new object and will have to be reloaded).